src/util/emitUtils.js

```diff
@@ -578,22 +580,28 @@ function emitToClients(io, nsp, rooms, key, obj) {
  */
 function doEmit(nsp, key, obj) {
   const newObjectAsString = getNewObjAsString(key, obj); // { key: {new: obj }}
-  const numClients = Object.values(nsp.connected).length;
+  const numClients = Object.values(nsp.rooms).length ?
+    Object.values(nsp.rooms).length : Object.values(nsp.connected).length;
   if (key.startsWith('refocus.internal.realtime.sample')) {
     tracker.trackEmit(obj.name, obj.updatedAt, numClients);
   }
   pubSubStats.trackEmit(key, obj);
-  if (!toggle.isFeatureEnabled('enableClientStats')) { // dont enter the else if its not a sample
-    nsp.emit(key, newObjectAsString);
-  } else if (key.startsWith('refocus.internal.realtime.sample')) {
-    Object.values(nsp.connected)
-      .forEach((socket) => {
+  if (toggle.isFeatureEnabled('enableClientStats') && key.startsWith('refocus.internal.realtime.sample')) { // dont enter the else if its not a sample
+    const rooms = nsp.rooms.slice(0);
+    nsp.rooms = [];
+    rooms
+      .forEach((room) => {
+        // Get socket for client who's id matches room
+        const socket = Object.values(nsp.connected)
+          .filter((connected) => Object.keys(connected.rooms)[1] === room)[0];
```

Comment:
There can be multiple sockets per room, so doing .filter(…)[0] means we would be skipping lots of sockets. You would need to do another forEach on the sockets that pass the filter.
But I think you can restructure it to use a single loop:
```
Object.values(nsp.connected)
.filter((socket) => nsp.rooms.includes(Object.keys(connected.rooms)[1]))
.forEach((socket) => ... )
```

Reply:
Ah nice, missed that. Thanks! Updated now.